src/raftstore/store/fsm/peer.rs

```diff
@@ -1275,142 +1377,189 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             return;
         }
 
-        if let Err(e) = self.schedule_merge(&region) {
-            info!(
-                "[region {}] failed to schedule merge, rollback: {:?}",
-                region.get_id(),
-                e
-            );
-            self.rollback_merge(&region);
-        }
-        self.merging_regions.as_mut().unwrap().push(region);
+        self.on_check_merge();
```

Comment:
IMO, `on_check_merge` only should called by timer, because it will register further timer in it.

Reply:
It will require keeping a useless tick running.